### DISCUSSION TITLE
OpenAI Network Problem on OpenWebUI

### DISCUSSION THREAD

**Discussion by whateverzpy** [open-webui/open-webui#3350] [General]
# Bug Report

## Description

**Bug Summary:**

I am experiencing an issue with OpenWebUI where it reports an "Openai: Network Problem" when setting the OpenAI API key. The same API key works fine with curl on my local machine.

```shell
curl http://localhost:8000/v1/chat/completions -H "Content-Type: application/json" -H "Authorization: Bearer sk-***************" -d '{"model": "gpt-4o","messages": [{"role": "user", "content": "Say this is a test!"}]}'
```

```shell
{"id":"chatcmpl-78823c3892804d50a3ec215253fb5dd4","model":"*************","object":"chat.completion","created":1718955051,"choices":[{"index":0,"message":{"role":"assistant","content":"# This is a Test!\n\nWelcome! How can I assist you with your questions or interests in Chinese modern and contemporary history today?"},"finish_reason":"stop"}],"usage":{"prompt_tokens":0,"completion_tokens":0,"total_tokens":0}}
```

**Steps to Reproduce:**

1. Deploy OpenAI API key on local server.
2. Use curl on local machine to verify API key works.
3. Set the same API key in OpenWebUI.

**Actual Behavior:**

OpenWebUI reports "Openai: Network Problem".

## Environment

- **Open WebUI Version:** 0.3.1

- **Operating System:** Windows 11
- **Browser (if applicable):** Edge 126.0.2592.61

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [ ] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**

![image](https://github.com/open-webui/open-webui/assets/86581757/ee4af7ea-e0cb-4436-a22e-426b12c11c3c)
![image](https://github.com/open-webui/open-webui/assets/86581757/202fa7a3-56ae-4d67-9b67-8af0a7d48ed1)
![image](https://github.com/open-webui/open-webui/assets/86581757/e700ef90-a186-4164-b190-64f4f7449c55)
Ignore the Ollama error because I'm not focusing on it.

**Docker Container Logs:**

```txt
2024-06-21 15:24:17 INFO:apps.openai.main:get_all_models()
2024-06-21 15:24:17 ERROR:apps.openai.main:Connection error: Cannot connect to host localhost:8000 ssl:default [Connection refused]
2024-06-21 15:24:17 INFO:apps.ollama.main:get_all_models()
2024-06-21 15:24:17 ERROR:apps.ollama.main:Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
2024-06-21 15:24:17 INFO:     127.0.0.1:42946 - "GET /health HTTP/1.1" 200 OK
2024-06-21 15:24:47 INFO:apps.openai.main:get_all_models()
2024-06-21 15:24:47 ERROR:apps.openai.main:Connection error: Cannot connect to host localhost:8000 ssl:default [Connection refused]
2024-06-21 15:24:47 INFO:apps.ollama.main:get_all_models()
2024-06-21 15:24:47 ERROR:apps.ollama.main:Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
2024-06-21 15:24:47 INFO:     127.0.0.1:51980 - "GET /health HTTP/1.1" 200 OK
2024-06-21 15:25:17 INFO:apps.openai.main:get_all_models()
2024-06-21 15:25:17 ERROR:apps.openai.main:Connection error: Cannot connect to host localhost:8000 ssl:default [Connection refused]
2024-06-21 15:25:17 INFO:apps.ollama.main:get_all_models()
2024-06-21 15:25:17 ERROR:apps.ollama.main:Connection error: Cannot connect to host host.docker.internal:11434 ssl:default [Connection refused]
2024-06-21 15:25:17 INFO:     127.0.0.1:48388 - "GET /health HTTP/1.1" 200 OK
```

## Installation Method

Docker

**Comment by mwmdev** [comment]
I'm having the same issue. I get his in the console when clicking on « Verify connexion » next to the OpenAI API:

![image](https://github.com/open-webui/open-webui/assets/31964517/9bd378f5-1baf-4fc3-add1-b0b53b3b17a0)


NixOS
Open WebUI v0.3.5
Ollama v0.1.28
Firefox 125

**Comment by Doutianbao** [comment]
same bug  on my docker machine.
any fix?

**Comment by Doutianbao** [comment]
![image](https://github.com/user-attachments/assets/4f4e2795-cfe9-4ec9-b618-d332aafb4167)

**Comment by syhya** [comment]
I also encountered the same error
<img width="1221" alt="Screenshot 2024-09-28 at 00 08 36" src="https://github.com/user-attachments/assets/49eca461-b0a0-427c-bfef-1deca98dcca0">

**Comment by Peteryhs** [comment]
I actually also encountered this issue. (Can't seem to ping any API backend and can't chat due to some json error. 

### Investigation:
I created different containers, each pointing to different data directories, and found that the issue stemmed from the database. Specifically, the application worked again when I manually copied over certain tables from the original database into a fresh instance.

### Resolution:
To resolve this issue:
1. I copied the following tables from the existing `webui.db` located in the data directory using DB browser for SQlite:
   - `user`
   - `chat`
   - `models`
   - `auth`
2. I created a new container and replaced its database tables with the copied tables.
3. This restored functionality to the application, and it began working as expected again.

I have no idea how safe this is, but it is working for now. Hoping for an update to officially address it!

**Comment by incohua** [comment]
I also encountered this problem, but I found the answer in the official [discussion](https://github.com/open-webui/open-webui/issues/3173).
I think I may have confused the container with the external network settings of the container.
You can access the service through ``host.decker.internal``
<img width="910" alt="{B705A00B-8580-48CE-8B78-A4B8617AF7CA}" src="https://github.com/user-attachments/assets/96a8ec36-e732-4063-8dc5-ee122fb74bfe">

**Reply by PeteHaughie** [reply]
This fixed it for me too. Point to your internal docker host and don't switch up the ports.

**Comment by c0gni** [comment]
I also had this problem, I check where problem can exits, focus on profile data, when u set folder in dokcer-compose to another path, where openwebui must save files, problem will resolve, but, what next, i will check.
